Doc/library/typing.rst

```diff
@@ -849,6 +849,22 @@ using ``[]``.
       concat(b"foo", b"bar")  # OK, output has type 'bytes'
       concat("foo", b"bar")   # Error, cannot mix str and bytes
 
+   Note that, despite its name, ``AnyStr`` has nothing to do with the
+   :class:`Any` type, nor does it mean "any string". In particular, ``AnyStr``
+   and ``str | bytes`` are different from each other and have different use
+   cases. If ``AnyStr`` is not part of the input arguments it should not be
+   used::
```

Comment:
Hmm. I just realised that this last sentence is not, strictly speaking, correct :/

This, for example, is a valid use of `AnyStr`:

```py
class Foo(Generic[AnyStr]):
    def method(self) -> AnyStr: ...
```

Note that `AnyStr` is not "part of the input arguments" in `method`'s signature there, but this is still a valid use of `AnyStr` as it's in the context of a class that is generic around `AnyStr`.

Maybe we should just remove this final sentence -- what do you think? I feel like the example below explains pretty well what we're talking about

```suggestion
   cases::
```

Reply:
Good catch and agreed that the examples explain themselves pretty well.